.github/workflows/core-pr-tests.yml

```
@@ -0,0 +1,96 @@
+    name: Run Core Provider Tests
+    
+    steps:
+      # Checkout the PR branch
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+      
+      # Set up Go environment
+      - name: Set up Go
+        uses: actions/setup-go@v4
```

actionlint (1.7.7)

the runner of "actions/setup-go@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)